src/index.ts

```diff
@@ -0,0 +1,83 @@
+interface IFigure {
+  readonly __name: string;
+  readonly color: string;
+  calculateArea(): number;
+}
+interface IAreaCalculation {
+  print(): string;
+}
+abstract class Figure implements IFigure {
+  constructor(
+    readonly __name: string,
+    readonly color: string
+  ) {}
+
+  abstract calculateArea(): number;
+}
+
+abstract class AreaCalculationFigure extends Figure implements IFigure, IAreaCalculation {
+  abstract print(): string;
+}
+
+class Circle extends Figure {
+  radius: number;
+  constructor(radius: number, color: string) {
```

Comment:
You are able to add a 'public' access modifier to 'radius' and remove everything else about radius.